src/components/general/button/index.tsx

```diff
@@ -4,17 +4,27 @@ import { IconProp } from '@fortawesome/fontawesome-svg-core';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { ContextualColors } from '../../../styles.d';
 
-const StyledButton = styled.button<{ color: keyof ContextualColors }>`
+const StyledButton = styled.button<{
+  color: keyof ContextualColors;
+}>`
   width: 100%;
   height: fit-content;
-  background: ${({ theme, color }) => theme.palette[color].main};
+  background: ${({ theme, color }) => theme.paletteV2[color].background};
   color: ${({ theme, color }) => theme.palette[color].contrast};
+  box-shadow: 3px 3px ${({ color, theme }) => theme.paletteV2[color].shadow};
+  border: 1px solid ${({ color, theme }) => theme.paletteV2[color].borderColor};
+  border-radius: ${({ color, theme }) => theme.paletteV2[color].borderRadius};
   padding: ${({ theme }) => theme.sizes.spacing.md};
   font-size: ${({ theme }) => theme.sizes.fonts.md};
   outline: none;
-  border: 1px solid ${({ theme, color }) => theme.palette[color].contrast};
   cursor: pointer;
 
+  :hover {
+    background: ${({ theme, color }) => theme.paletteV2[color].hover};
+    box-shadow: 3px 3px
+      ${({ color, theme }) => theme.paletteV2[color].hoverShadow};
+    color: ${({ theme }) => theme.palette.primary.main};
```

Comment:
Should there maybe be a main and contrast color on paletteV2 so all the colors come from the same palette?